bindata/network/ovn-kubernetes/self-hosted/alert-rules.yaml

```diff
@@ -0,0 +1,383 @@
+        != count(max_over_time(kube_node_role{role="master"}[5m])) * (count(max_over_time(kube_node_role{role="master"}[5m]))-1)
+      for: 5m
+      labels:
+        severity: warning
+    - alert: OVNKubernetesSouthboundDatabaseOutboundConnectionMissing
+      annotations:
+        summary: OVN southbound database server(s) do not have expected number of outbound RAFT connections.
+        description: OVN southbound database server(s) do not have expected number of outbound connections for a RAFT cluster
+          which may indicate degraded OVN database high availability.
+      expr: |
+        # Expected sum of outbound connections is number of control plane nodes * number of control plane nodes minus one
+        # Without max_over_time, failed scrapes could create false negatives, see
+        # https://www.robustperception.io/alerting-on-gauges-in-prometheus-2-0 for details.
+        sum(max_over_time(ovn_db_cluster_outbound_connections_total{db_name="OVN_Southbound"}[5m]))
+        != count(max_over_time(kube_node_role{role="master"}[5m])) * (count(max_over_time(kube_node_role{role="master"}[5m]))-1)
+      for: 5m
+      labels:
+        severity: warning
+    # OVN northbound and southbound database performance alerts
+    - alert: OVNKubernetesNorthboundDatabaseCPUUsageHigh
+      annotations:
+        summary: OVN northbound database {{"{{"}} $labels.instance {{"}}"}} is greater than {{"{{"}} $value {{"}}"}} percent CPU usage for a period of time.
+        description: High OVN northbound CPU usage indicates high load on the networking control plane.
+      expr: (sum(rate(container_cpu_usage_seconds_total{container="nbdb"}[5m])) BY (instance, name) * 100) > 80
+      for: 15m
+      labels:
+        severity: info
+    - alert: OVNKubernetesSouthboundDatabaseCPUUsageHigh
+      annotations:
+        summary: OVN southbound database {{"{{"}} $labels.instance {{"}}"}} is greater than {{"{{"}} $value {{"}}"}} percent CPU usage for a period of time.
+        description: High OVN southbound CPU usage indicates high load on the networking control plane.
+      expr: (sum(rate(container_cpu_usage_seconds_total{container="sbdb"}[5m])) BY (instance, name) * 100) > 80
+      for: 15m
+      labels:
+        severity: info
+    # OVN northd functional alerts
+    - alert: OVNKubernetesNorthdInactive
+      annotations:
+        summary: No OVN northd is active. Networking is degraded.
+        description: No OVN northd is available within the high availability set. Networking control plane is degraded.
+      expr: |
+        # Without max_over_time, failed scrapes could create false negatives, see
+        # https://www.robustperception.io/alerting-on-gauges-in-prometheus-2-0 for details.
+        count(max_over_time(ovn_northd_status[5m]) == 1) != 1
+      for: 5m
+      labels:
+        severity: critical
+    # OVN kubernetes node functional alerts
+    - alert: NodeWithoutOVNKubeNodePodRunning
```

Comment:
The alerts below will be deployed to hosted clusters only?
Seems at least this one `NodeWithoutOVNKubeNodePodRunning` is also useful for management cluster.

Reply:
The alerts here are only deployed on the management clusters and not the hosted clusters.

Reply:
As my understanding the yaml defined in `self-hosted` will be deployed to hosted clusters? Could you help double confirm?

Reply:
@zshi-redhat My understanding is the managed alerts are for the hosted clusters and the self hosted alerts will be used in the management cluster. Correct?

Reply:
@martinkennelly  
The `managed` alerts in directory managed are for the hosted clusters - yes
The `self-hosted` alerts in directory self-hosted are for non-hypershift openshift deployment, which can be used in the management cluster in management cluster is using ovnk network type.